src/components/ui/calendar.tsx

```diff
@@ -0,0 +1,213 @@
+
+function Calendar({
+  className,
+  classNames,
+  showOutsideDays = true,
+  events = [],
+  ...props
+}: CalendarProps & { events: EventProps[] }) {
+  const [selectedDate, setSelectedDate] = React.useState<Date | undefined>(
+    undefined,
+  );
+  const [popupEvent, setPopupEvent] = React.useState<EventProps | null>(null);
+
+  const eventDays = events.map((event) => event.date);
+
+  const modifiers = {
+    hasEvent: (date: {
+      getFullYear: () => number;
+      getMonth: () => number;
+      getDate: () => number;
+    }) =>
+      eventDays.some(
+        (eventDate) =>
+          date.getFullYear() === eventDate.getFullYear() &&
+          date.getMonth() === eventDate.getMonth() &&
+          date.getDate() === eventDate.getDate(),
+      ),
+  };
+
+  const CustomDay: React.FC<{
+    date: Date;
+    selected: boolean;
+    onClick: (date: Date) => void;
+  }> = ({ date, selected, onClick }) => {
+    const today = new Date();
+    const isToday =
+      date.getDate() === today.getDate() &&
+      date.getMonth() === today.getMonth() &&
+      date.getFullYear() === today.getFullYear();
+
+    const dayEvents = events.filter(
+      (event) =>
+        event.date.getFullYear() === date.getFullYear() &&
+        event.date.getMonth() === date.getMonth() &&
+        event.date.getDate() === date.getDate(),
+    );
+
+    return (
+      <div className="relative overflow-clip text-[1vw]">
+        <div
+          onClick={() => onClick(date)}
+          className={cn(
+            buttonVariants({ variant: "ghost" }),
+            "h-[10vw] w-[9.7vw] px-1 py-0 text-[2vw] aria-selected:bg-gray-100 aria-selected:text-black md:h-[7vw] md:text-[1vw]",
+            selected
+              ? "bg-gray-100 text-black hover:bg-gray-50 hover:text-black"
+              : "",
+            isToday ? "bg-gray-200" : "",
+          )}
+        >
+          {date.getDate()}
+        </div>
+
+        {dayEvents.length > 0 && (
+          <div className="absolute left-[5%] top-[28%] w-[90%]">
+            {dayEvents.map((event, index) => (
+              <div
+                key={index}
+                onClick={() => setPopupEvent(event)}
+                className="mb-0 cursor-pointer rounded-lg border border-black text-[0.9vw]"
+              >
+                <div className="truncate px-[3%]">
+                  {new Date(event.startTime).toLocaleTimeString([], {
+                    hour: "2-digit",
+                    minute: "2-digit",
+                  })}{" "}
+                  {event.title}
+                </div>
+              </div>
+            ))}
+          </div>
+        )}
+      </div>
+    );
+  };
+
+  return (
+    <div>
+      <DayPicker
+        showOutsideDays={showOutsideDays}
+        className={cn("p-0", className)}
+        modifiers={modifiers}
+        classNames={{
+          month: "space-y-[0.5vw]",
+          caption:
+            "justify-center flex pl-[80vh] pt-[0.5vh] items-center relative mb-[3vh]",
```

Comment:
```suggestion
            "justify-center flex pl-[42.5vw] pt-[0.5vh] items-center relative mb-[3vh]",
```

you'd want to use vw here since this is horizontal positioning. you can adjust the value to better center it too.